nncf/openvino/graph/layer_attributes.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional
+
+from nncf.common.graph.layer_attributes import BaseLayerAttributes
+from nncf.common.graph.layer_attributes import ConvolutionLayerAttributes
+from nncf.common.graph.layer_attributes import GenericWeightedLayerAttributes
+from nncf.common.graph.layer_attributes import WeightedLayerAttributes
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVConvolutionMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVDepthwiseConvolutionMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVGroupConvolutionMetatype
+
+
+class OVLayerAttributes(BaseLayerAttributes):
+    """
+    This class stores additional information about nodes that needs to be processed during compression.
+    """
+
+    def __init__(
+        self,
+        const_attrs: Dict[int, Any],
+        common_layer_attrs: Optional[Dict[int, BaseLayerAttributes]] = None,
+        act_attrs: Optional[Dict[Any, Any]] = None,
+    ):
+        """
+        :param const_attrs: Map of weights port ID to corresponding const attributes.
+        :param act_attrs: Activation attributes.
+        """
+        self._const_attrs = const_attrs
+        self._common_layer_attrs = {} if common_layer_attrs is None else common_layer_attrs
+        self._act_attrs = {} if act_attrs is None else act_attrs
```

Comment:
Could you please explain why this does not work?
```python
self._common_layer_attrs = common_layer_attrs
self._act_attrs = act_attrs

@property
def const_attrs(self) -> Dict[int, Any]:
    return self._const_attrs

@property
def act_attrs(self) -> Optional[Dict[Any, Any]]:
    return self._act_attrs
```

Reply:
```python
def get_const_port_ids(self) -> List[int]:
    if self._const_attrs:
        return list(self._const_attrs.keys())
    return []
```

Reply:
This is copypast, I'll fix this